controllers/reconcilers/grafana/ingress_reconciler.go

```diff
@@ -36,12 +36,13 @@ func NewIngressReconciler(client client.Client, isOpenShift bool) reconcilers.Op
 func (r *IngressReconciler) Reconcile(ctx context.Context, cr *v1beta1.Grafana, vars *v1beta1.OperatorReconcileVars, scheme *runtime.Scheme) (v1beta1.OperatorStageStatus, error) {
 	log := logf.FromContext(ctx).WithName("IngressReconciler")
 
-	if r.isOpenShift {
-		log.Info("reconciling route", "platform", "openshift")
-		return r.reconcileRoute(ctx, cr, vars, scheme)
-	} else {
+	// Prefer Ingress on Openshift when spec.ingress is defined
+	if !r.isOpenShift || (cr.Spec.Ingress != nil && cr.Spec.Route == nil) {
```

Comment:
I guess something like this would be better style-wise (just typing from my phone, so might not be entirely correct):

```go
if r.isOpenshift && cr.Spec.Route != nil {
    log.Info("reconciling route", "platform", "openshift")
    return r.reconcileRoute(ctx, cr, vars, scheme)
}

log.Info("reconciling ingress")
return r.reconcileIngress(ctx, cr, vars, scheme)
```

Reply:
What do you think about simplifying if condition like in my example? I think "if r.isOpenshift && cr.Spec.Route != nil" is much easier to read than "if !r.isOpenShift || (cr.Spec.Ingress != nil && cr.Spec.Route == nil)"